### PR TITLE
fix #217 and add shifting to the tree

### LINKED ISSUE #217: resource_path is not working
Basically [this line](https://github.com/biocore/empress/blob/master/empress/core.py#L309,) should be:
`'base_url': './support_files',` -> `'base_url': self.base_url,`

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: empress/tools.py [hunk 1]
```diff
@@ -273,3 +274,45 @@ def match_inputs(
             )
 
     return ff_table, sf_sample_metadata, tip_metadata, int_metadata
+
+
+def shifting(bitlist, size=51):
+    """Takes a list of 0-1s, splits in in size and converts it to a list of int
```
Comment (ElDeveloper) [inline empress/tools.py:280]:
```suggestion
    """Takes a list of 0-1s, splits in size and converts it to a list of int
```

File: empress/tools.py [hunk 2]
```diff
@@ -273,3 +274,45 @@ def match_inputs(
...
+    ----------
+    bitlist: list of int
+        The input list of 0-1
+    size: int
+        The size of the buffer
+
+    Returns
+    -------
+        list of int
+            Representation of the 0-1s as a list of int
```
Comment (ElDeveloper) [inline empress/tools.py:292]:
```suggestion
    list of int
        Representation of the 0-1s as a list of int
```

File: tests/python/test_tools.py [hunk 3]
```diff
@@ -392,6 +392,30 @@ def test_match_inputs_feature_metadata_only_internal_node_metadata(self):
...
+            ([1], [1]),
+        ]
+
+        for test, obs in tests:
+            self.assertEqual(tools.shifting(test), obs)
+            self.assertEqual(_count_bits(obs[0]), len(test))
+
+        with self.assertRaisesRegex(ValueError, "Your list has values other "
+                                    "than 0-1s"):
+            tools.shifting([10])
```
Comment (ElDeveloper) [inline tests/python/test_tools.py:417]: 😆

File: empress/support_files/js/bp-tree.js [hunk 4]
```diff
@@ -12,12 +12,34 @@ define(["ByteArray"], function (ByteArray) {
...
          * @type {Uint8Array}
          * Used to store the structure of the tree
          * @private
          */
+
+        if (coding !== undefined) {
+          var b_len = b.length - 1;
+
+          var decoded_b = [];
+          _.each(b, function (value, i) {
```
Comment (ElDeveloper) [inline empress/support_files/js/bp-tree.js:26]: Can you add a test for this block in `test-bp-tree.js`?

File: empress/core.py [hunk 5]
```diff
@@ -104,7 +104,7 @@ def __init__(self, tree, table, sample_metadata,
 
         self.base_url = resource_path
         if self.base_url is None:
-            self.base_url = './'
+            self.base_url = 'support_files'
```
[the remark this answers is not in the document]
Comment (antgonza) [reply empress/core.py:107]: Yes, this is an interesting one. I _think_ that we need to make support_files the default so this works well with Qiime2 but I _might_ be wrong. Either way, I _think_ that is better to simply pass the folder where that file exist vs the fullpath; like what is done in Emperor with the "emperor files".

File: empress/support_files/js/bp-tree.js [hunk 6]
```diff
@@ -8,16 +8,49 @@ define(["ByteArray"], function (ByteArray) {
...
      * @param {Array} names The names of each node stored in preorder
      * @param {Array} lengths The lengths of each node stored in preorder
+     * @param {Number} coding The number of 1/0s coded in the tree, null not coded
      *
      * @return {BPTree}
      * @constructs BPTree
      */
-    function BPTree(b, names = null, lengths = null) {
+    function BPTree(b, names = null, lengths = null, coding = 51) {
         /**
```
Comment (ElDeveloper) [inline empress/support_files/js/bp-tree.js:17]: Can you move this docstring to be next to the declaration of `this.b_` in line 54?

File: empress/tools.py [hunk 7]
```diff
@@ -273,3 +274,50 @@ def match_inputs(
...
+    list of int
+        Representation of the 0-1s as a list of int
+
+    Raises
+    ------
+    ValueError
+        If any of the list values is different than 0 or 1
+
+
+    The tree to be visualized.
```
Comment (ElDeveloper) [inline empress/tools.py:300]:
```suggestion
```

File: tests/test-bp-tree.js [hunk 8]
```diff
@@ -258,5 +258,27 @@ require(['jquery', 'ByteArray', 'BPTree'], function($, ByteArray, BPTree) {
                 equal(this.bpObj.preorderselect(k + 1), exp[k]);
             }
         });
+
+        test('Test coding', function() {
+          // zeros and ones test
+          obj = new BPTree([3851728]);
```
Comment (ElDeveloper) [inline tests/test-bp-tree.js:264]:
```suggestion
          var obj = new BPTree([3851728]);
```

File: tests/test-bp-tree.js [hunk 9]
```diff
@@ -258,5 +258,27 @@ require(['jquery', 'ByteArray', 'BPTree'], function($, ByteArray, BPTree) {
...
+          equal(obj.b, this.bpObj.b);
+
+          // zeros test
+          exp = [0, 0, 0, 0, 0, 0];
```
Comment (ElDeveloper) [inline tests/test-bp-tree.js:268]:
```suggestion
          var exp = [0, 0, 0, 0, 0, 0];
```

**Comment by ElDeveloper** [comment]
Thanks @antgonza! Looks like there's conflicts, can you resolve those? Also, @kwcantrell if this looks good to you can you go ahead and merge this PR?

**Comment by ElDeveloper** [comment]
@antgonza, there's conflicts again due to a merged PR. Can you re-solve conflicts?

**Comment by ElDeveloper** [comment]
@antgonza, let's wait for @kwcantrell to submit his review and then this should be good to merge and solve conflicts.

File: empress/support_files/js/bp-tree.js [hunk 10]
```diff
@@ -8,11 +8,44 @@ define(["ByteArray"], function (ByteArray) {
...
+                    }
+                    decoded_b.push.apply(decoded_b, element);
+                }
+            });
+
+            b = decoded_b;
+        }
+
         /**
          * @type {Uint8Array}
```
Comment (kwcantrell) [inline empress/support_files/js/bp-tree.js:50]:
```suggestion
         * @type {Array}
```

**Comment by ElDeveloper** [comment]
Thanks @antgonza @kwcantrell!

**Pull request merged**